File: Messages/Database/Column.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Messages.Database
{
    public partial class Column
    {
        public Column(string name, string type, string[] mods, bool primaryKey)
        {
            this.name = name.ToLower();
            this.type = type;
            this.primaryKey = primaryKey;
            this.mods = mods;
        }

        /// <summary>
        /// Returns the structure of the column in a way that can be used in a
        /// CREATE table statement
        /// </summary>
        /// <returns>The structure of the column as a string</returns>
        public string getCreateStructure()
        {
            string structure = name + " " + type;
            if (mods != null)
            {
                foreach (string mod in mods)
                {
                    structure += " " + mod;
                }
            }
            return structure;
        }
    }

    public partial class Column
    {
        /// <summary>
        /// The name of the column
        /// </summary>
        private string name;
        public string getName()
        {
            return name;
        }

        /// <summary>
        /// The data type of the column
        /// </summary>
        private string type;
        public string getType()
        {
            return type;
        }

        /// <summary>
        /// The modfiers on the column, such as UNIQUE or NOT NULL
        /// May be null
        /// </summary>
        private string[] mods;
        public string[] getMods()
        {
            return mods;
        }

        /// <summary>
        /// Represents whether or not this column is a primary key.
        /// </summary>
        private bool primaryKey;
        public bool isPrimary()
        {
            return primaryKey;
        }
    }
}

File: CompanyReviewService_Java/src/companyReviewService/database/Column.java
package companyReviewService.database;

import java.util.ArrayList;

/**
 * This class represents a single column in a MySQL database table
 * @author joshua
 */
public class Column {
	
	public Column(String name, String type, ArrayList<String> mods, boolean pKey){
		this.name = name;
		this.type = type;
		this.mods = mods;
		this.pKey = pKey;
	}
	
	/**
	 * Generates a string that can be used in a MySQL CREATE query to create this column.
	 * @return string used in CREATE statements
	 */
	public String getCreateStructure(){
		String structure = name + " " + type;
		if(mods != null){
			for(int i = 0; i != mods.size(); i++){
				structure += " " + mods.get(i);
			}
		}
		return structure;
	}
	
	/**
	 * The name of the column
	 */
	private String name;
	public String getName(){ return name; }
	
	/**
	 * The datatype of the column
	 */
	private String type;
	public String getType(){ return type; }
	
	/**
	 * Any modifications to the columns, such as NOT NULL or UNIQUE
	 */
	private ArrayList<String> mods;
	public ArrayList<String> getMods(){ return mods; }
	
	/**
	 * Indicates whether or not this column is a primary key
	 */
	private boolean pKey;
	public boolean isPKey(){ return pKey; }
}
